tests/unit/utils/test_templates.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+'''
+Unit tests for salt.utils.templates.py
+'''
+
+# Import python libs
+from __future__ import absolute_import, print_function, unicode_literals
+import sys
+import logging
+
+# Import Salt libs
+import salt.utils.templates
+
+# Import Salt Testing Libs
+from tests.support.unit import TestCase, skipIf
+
+log = logging.getLogger(__name__)
+
+
+### Default context for salt.utils.templates.render_*_tmpl to work
+context = {
```

Comment:
Don't set this dictionary at the global module level.
If you only want it generated once, add it as `cls.context` in the `setUpClass` classmethod.

Reply:
Though, adding it as `self.context` in the `setUp` method will remove the need for your `dict(context)` calls.

```python
def setUp(self):
    self.context = {the dict contents}
    self.addCleanup(delattr, self, 'context')
```

Reply:
kk. I'll move that then in a sec.

Reply:
yeah, resuming my vim session it seems that I originally did that global setting of the options because it seems a number of the units tests (specifically test_gitfs which this was mostly copied from for form) assigns the options globally, but yeah.

I'll have this commit in a sec once the tests complete.

Reply:
Yes, they do, and they will stop doing it :smile: